expect/_utils.ts

```diff
@@ -18,3 +18,82 @@ export function isPromiseLike(value: unknown): value is PromiseLike<unknown> {
     return typeof ((value as Record<string, unknown>).then) === "function";
   }
 }
+
+// SENTINEL constants are from https://github.com/facebook/immutable-js
+const IS_KEYED_SENTINEL = "@@__IMMUTABLE_KEYED__@@";
+const IS_SET_SENTINEL = "@@__IMMUTABLE_SET__@@";
+const IS_LIST_SENTINEL = "@@__IMMUTABLE_LIST__@@";
+const IS_ORDERED_SENTINEL = "@@__IMMUTABLE_ORDERED__@@";
+const IS_RECORD_SYMBOL = "@@__IMMUTABLE_RECORD__@@";
```

Comment:
Checks for these sentinel keys/values don't make much sense to me because nothing seems setting these keys.

I guess we can just remove these constants and checks related to these

Reply:
For internal codebase, it is yes. But as for the apps which uses `immutable.js`, they will rely on these keys/value to test it, such as the following:

```ts
import { OrderedMap } from "npm:immutable@4.3.5";
import { expect, iterableEquality } from "https://deno.land/std@$STD_VERSION/expect/mod.ts";

Deno.test("immutable test", () => {
    const a = OrderedMap().set("saving", true);
    const b = OrderedMap().merge({saving: true});
    expect(iterableEquality(a, b)).toBe(true);
});
```

Reply:
Does jest support such cases natively?

Reply:
Yes, the `iterableEquality` is exported from `expect-utils`, and the test codebase is [here](https://github.com/jestjs/jest/blob/main/packages/expect-utils/src/__tests__/utils.test.ts#L579)